### PR TITLE
[clang-doc][nfc] Avoid constructing SmallString in ToString method

### PR DESCRIPTION
**Pull request by ilovepi**
This patch updates the return type of HTMLTag::toString from
SmallString<16> to StringRef, since this API only returns string
literals. As a result, there is no need to construct a full heap
allocated or owned string.

Additionally, this patch renames ToString to toString to match the LLVM
style guide.

### CODE REVIEW

**Comment by ilovepi** [comment]
@PeterChou1 please take a look and provide some feedback.

File: clang-tools-extra/clang-doc/HTMLGenerator.cpp [hunk 1]
```diff
@@ -56,7 +56,7 @@ class HTMLTag {
   operator bool() = delete;
 
   bool IsSelfClosing() const;
-  llvm::SmallString<16> ToString() const;
+  const char* toString() const;
```
Comment (ilovepi) [inline clang-tools-extra/clang-doc/HTMLGenerator.cpp:59]: Maybe this should be renamed to `c_str()` or `str()`?

Comment (PeterChou1) [reply clang-tools-extra/clang-doc/HTMLGenerator.cpp:59]: I think it be more appropriate to rename it c_str

Comment (ilovepi) [reply clang-tools-extra/clang-doc/HTMLGenerator.cpp]: done

**Review by PeterChou1 — APPROVED** [review]

**Comment by petrhosek** [comment]
I'd use `std::string_view` as a return value instead of `const char *` as a more idiomatic C++ in which case you don't need to rename the method to `c_str()`.

**Comment by ilovepi** [comment]
> I'd use `std::string_view` as a return value instead of `const char *` as a more idiomatic C++ in which case you don't need to rename the method to `c_str()`.

I opted for `StringRef`, since that seems more in line w/ the conventions here, but I'm happy to switch to `std::string_view` if you'd prefer.

**Comment by petrhosek** [comment]
> > I'd use `std::string_view` as a return value instead of `const char *` as a more idiomatic C++ in which case you don't need to rename the method to `c_str()`.
> 
> I opted for `StringRef`, since that seems more in line w/ the conventions here, but I'm happy to switch to `std::string_view` if you'd prefer.

AFAIK the long-term plan is to replace `StringRef` with `std::string_view` but I'm fine using `StringRef` for consistency with the rest of the codebase.

**Review by petrhosek — APPROVED** [review]

**Pull request merged**